Libraries/LibAudio/ABuffer.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <AK/RefCounted.h>
+#include <AK/ByteBuffer.h>
+#include <AK/Types.h>
+#include <AK/Vector.h>
+
+// A single sample in an audio buffer.
+// Values are floating point, and should range from -1.0 to +1.0
+struct ASample {
+    ASample()
+        : left(0)
+        , right(0)
+    {}
+
+    // For mono
+    ASample(float left)
+        : left(left)
+        , right(left)
+    {}
+
+    // For stereo
+    ASample(float left, float right)
+        : left(left)
+        , right(right)
+    {}
+
+    void clamp()
+    {
+        if (left > 1)
+            left = 1;
+        else if (left < -1)
+            left = -1;
+
+        if (right > 1)
+            right = 1;
+        else if (right < -1)
+            right = -1;
+    }
+
+    ASample& operator+=(const ASample& other)
+    {
+        left += other.left;
+        right += other.right;
+        return *this;
+    }
+
+    float left;
+    float right;
+};
+
+// A buffer of audio samples, normalized to 44100hz.
+class ABuffer : public RefCounted<ABuffer> {
+public:
+    static RefPtr<ABuffer> from_pcm_data(ByteBuffer& data, int num_channels, int bits_per_sample, int source_rate);
+    ABuffer(Vector<ASample>& samples)
```

Comment:
Let's try to avoid a `Vector` copy here, since it's so easy :)
The most straightforward way would be to take a `Vector<ASample>&&` to force call sites to `move()` it, then just `move()` it into `m_samples` below.